File: CPP VW/CPP VW/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(World& world, int x, int y) :
	Plant(world, x, y)
{
	this->sign = 'x';
	this->power = 0;
	this->initiative = 0;
}
bool Guarana::specialCollision(Organism* other) {
	other->SetPower(other->GetPower() + 3);
	return false;
}
bool Guarana::specialAction() {
	return false;
}
Guarana::~Guarana() {
}

File: JAVA VW/src/programowanie/obiektowe/proj/Guarana.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package programowanie.obiektowe.proj;

/**
 *
 * @author kubaz
 */
public class Guarana  extends Plant{
    
    public Guarana(Position position, World world) {
        super(position, world);
        this.name="Guarana";
        this.initiative=0;
        this.strength=0;
    }
    
    @Override
    public boolean specialCollision(Organism other)
    {
        other.setStrength(other.getStrength()+3);
        return true;
    }
    
}
